app/waf/src/application.ts

```diff
@@ -51,5 +56,61 @@ export class WafApplication extends BootMixin(
         nested: true,
       },
     };
+
+    if (process.env.TEST_OPENSTACK_INTEGRATION)
+      this.controller(OpenstackController);
+  }
+
+  async keepAliveAdminToken() {
+    if (!process.env.PRODUCT_RELEASE) return;
```

Comment:
What does it mean?

Reply:
I don't want the auth process to bother us during the developing. So this is an on-off if we really need to connect OpenStack for auth*.